web-marketplace/src/components/organisms/RegistryLayout/RegistryLayout.Header.tsx

```diff
@@ -121,6 +122,15 @@ const RegistryLayoutHeader: React.FC = () => {
     ? headerColors[pathname]
     : theme.palette.primary.light;
 
+  const onLogOut = async () => {
+    await disconnect();
+
+    // Remove all localStorage items related with the buy credits flow
+    Object.keys(localStorage)
+      .filter(key => key.startsWith(BUY_CREDITS_FORM_PREFIX))
+      .forEach(key => localStorage.removeItem(key));
+  };
```

Comment:
what about adding this logic to the end of the `disconnect` function itself?
Right now we only have 1 log out button in the app but if we want to add more in the future for some reason, it could be useful to have this more integrated